spanner/spanner_snippets/snippet.go

```diff
@@ -477,6 +483,234 @@ func writeWithTransactionUsingDML(ctx context.Context, w io.Writer, client *span
 
 // [END spanner_dml_getting_started_update]
 
+// [START spanner_postgresql_create_database]
+
+func pgCreateDatabase(ctx context.Context, w io.Writer, adminClient *database.DatabaseAdminClient, db string) error {
+	matches := regexp.MustCompile("^(.*)/databases/(.*)$").FindStringSubmatch(db)
+	if matches == nil || len(matches) != 3 {
+		return fmt.Errorf("invalid database id %s", db)
+	}
+	op, err := adminClient.CreateDatabase(ctx, &adminpb.CreateDatabaseRequest{
+		Parent:          matches[1],
+		DatabaseDialect: adminpb.DatabaseDialect_POSTGRESQL,
+		// Note that PostgreSQL uses double quotes for quoting identifiers. This also
+		// includes database names in the CREATE DATABASE statement.
+		CreateStatement: `CREATE DATABASE "` + matches[2] + `"`,
+	})
+	if err != nil {
+		return err
+	}
+	if _, err := op.Wait(ctx); err != nil {
+		return err
+	}
+	// Databases that are created with PostgreSQL dialect do not support extra DDL statements in the `CreateDatabase` call.
```

Comment:
Please wrap this comment to under 100 characters, or even shorter (80). This will be hard to read on cloud.google.com. See https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#line-length.